src/driver/linux_resource/syscall_x86.c

```diff
@@ -253,39 +253,24 @@ void** find_syscall_table(void)
          (p[2] == 0xe7 && p[4] == 0x63 && p[5] == 0xf0)) ) {
       /* Skip IBRS: search for the nearest "call", 0xe8. */
       unsigned char *p1 = p + 6;
-      while( *p1 != 0xe8 )
-        p1++;
-      /* Is this our "call do_syscall_64", or something from UNTRAIN_RET?
-       * In the second case these calls go one after another, and we should
-       * take the second one.
-       *
-       * To get into this case, you need to pass retbleed=unret or
-       * retbleed=ibpb via the kernel command line.
-       * Tested with Intel(R) Xeon(R) CPU E3-1220 v6 @ 3.00GHz.
-       *
-       * No kernel I know about has a "call" after "call do_syscall_64",
-       * so 2 consecutive calls mean that do_syscall_64 is the second one. */
-      if( p1[5] == 0xe8 )
-        p1 += 5;
+      unsigned char *ret;
+      do {
+        while( *p1 != 0xe8 )
+          p1++;
+        result = (unsigned long)p1 + 5;
+        result += p1[1] | (p1[2] << 8) | (p1[3] << 16) | (p1[4] << 24);
 
-      result = (unsigned long)p1 + 5;
-      result += p1[1] | (p1[2] << 8) | (p1[3] << 16) | (p1[4] << 24);
-      break;
+        ret = find_syscall_table_from_do_syscall_64((void*)result);
+        if( ret != NULL )
+          return (void**)ret;
+        p1++;
```

Comment:
Not sure, but shall we skip five bytes instead of one if the address contains `0xe8` so that we don't call the `find_syscall_table_from_do_syscall_64()` predicate on data which is certainly not an address? Still, one byte doesn't look like a problem to me, but five seems a reasonable optimisation. What do you reckon?